### ISSUE TITLE
Should show diagnostic for `dataclass_transform` field specifier with `alias` that is not a valid identifier

### ISSUE THREAD

**Issue by debonte** [microsoft/pyright#9220]
**Describe the bug**
When using the `alias` argument on a `dataclass_transform` field specifier, if the value provided is not a valid identifier, I think we should show a diagnostic.

In https://github.com/microsoft/pylance-release/issues/6554, the user provided an alias with a space in it as shown in the example below. This results in the completion provider suggesting that invalid name as a parameter name.

![Image](https://github.com/user-attachments/assets/598a91b2-fa39-45e7-9872-45acec733111)

**Code or Screenshots**
pip install latest pydantic, and then:

```
from pydantic import BaseModel, Field

class SettingsModel(BaseModel):
    env: int = Field(alias='Invalid Identifier')
    
settingsMod = SettingsModel()
```

**VS Code extension or command-line**
Using Pylance in VS Code.

**Comment by erictraut** [comment]
I agree this would be a good check to add given that the runtime support generates an exception in this case.

This will be addressed in the next release.

**Comment by Viicos** [comment]
For consistency, should the same be applied for `Unpack` and `TypedDict`? ([playground](https://pyright-play.net/?pyrightVersion=1.1.384&strict=true&enableExperimentalFeatures=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgCYAiSAxjADRQCqKCAhlQNYBQnB5UAvIRIVqMABQAiXhPoBvCQEkUANxYAbJKSgLSxFDCTAkxEBIBcmfQF8AlN13AowAK4oqYgFQf2AdxYg0AGcLJlYOAG1eAF0bM04oBKhWQMDuFzcxGyA)). It's maybe not necessary as it might be more common to do `func(**{"Invalid Identifier": 1})` in that case.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.385.

**Comment by debonte** [comment]
Was this reopened intentionally? It has effectively been won't fixed.
